staking/slash/double-sign.go

```diff
@@ -143,17 +191,10 @@ func (r Record) String() string {
 	return string(s)
 }
 
-// CommitteeReader ..
-type CommitteeReader interface {
-	Config() *params.ChainConfig
-	ReadShardState(epoch *big.Int) (*shard.State, error)
-	CurrentBlock() *types.Block
-}
-
 // Verify checks that the slash is valid
 func Verify(
 	chain CommitteeReader,
-	state *state.DB,
+	state stateDB,
```

Comment:
Why is this not a pointer anymore

Reply:
*state.DB is abstracted to an interface. This will help me to write a fake state db only for testing.

Reply:
Please remove this interface. You should not change signatures just to make testing easier. Testing consideration should not affect logic in any way

Reply:
The interface is not just to make tests easier. Actually it's a practice of [dependency inversion principle](https://en.wikipedia.org/wiki/Dependency_inversion_principle).

Reply:
I see both of your points. But since we don't have interface in other places. It's better to not change the signature here (unless we need to change it everywhere). Is it complicated to instead use a Mock state.DB for testing?

Reply:
Gotta. Will treat this principle as our coding standard. Thanks. 👍

Reply:
stateDB reverted to *state.DB. Work done.